### PR TITLE
Fixvalidate

### PR DESCRIPTION
**Pull request by dat26061994**
To day, i fix validate input Form

### CODE REVIEW

File: app/Http/Controllers/ProductController.php [hunk 1]
```diff
@@ -84,19 +86,21 @@ public function postEdit($id, Request $request)
         $productModel = new Product;
         $this->validate($request,
```
Comment (kakeyang) [inline app/Http/Controllers/ProductController.php:90]: Validation rule should be written in Controller?

**Comment by kakeyang** [comment]
I see you committed some test image files but you don't need to do that.

File: app/Http/Controllers/AdminUserController.php [hunk 2]
```diff
@@ -75,12 +75,13 @@ public function postEdit($id, Request $request)
         $adminModel = new Admin;
         $this->validate($request,
             [
-                'txtName' => 'required',
+                'txtName' => 'required|regex:/(^[A-Za-z ]+$)+/',
```
Comment (4c0n) [inline app/Http/Controllers/AdminUserController.php:78]: Is txtName the name of a person? Be careful with this regex people can also have a "-" character in their name.

File: app/Http/Controllers/ProductController.php [hunk 3]
```diff
@@ -33,9 +33,11 @@ public function getAdd()
 
     public function postAdd(ProductRequest $request)
     {
-        $file_name = $request->file('fImages')->getClientOriginalName();
         $product = new Product;
 
+        $file_name = $product->rand_string(10) . '-' . $request->file('fImages')->getClientOriginalName();
```
Comment (4c0n) [inline app/Http/Controllers/ProductController.php]: Please use camelCase for variable names. I think phpcs will tell you the same.

File: app/Http/Controllers/ProductController.php [hunk 4]
```diff
@@ -84,19 +86,21 @@ public function postEdit($id, Request $request)
         $productModel = new Product;
         $this->validate($request,
             [
-                'txtName' => 'required|max:100',
-                Rule::unique('products')->ignore($productModel->id),
+                'txtName' => 'required|max:100|regex:/(^[A-Za-z0-9 ]+$)+/',
+                'txtName' => 'unique:products,name,' . $id,
                 'txtPrice' => 'required|numeric|max:10000000000',
-                'txtDescription' => 'max:300',
+                'txtDescription' => 'max:300|regex:/(^[A-Za-z ]+$)+/',
```
Comment (4c0n) [inline app/Http/Controllers/ProductController.php]: You can only have letters in the description? What about numbers, ".,-?!/", etc..?

File: app/product.php [hunk 5]
```diff
@@ -52,5 +52,16 @@ public function findProduct($id)
         return $findProduct;
     }
 
+    function rand_string($length)
```
Comment (4c0n) [inline app/product.php]: This function should probably be in a Helper class:
https://laravel.com/docs/5.4/helpers

**Review by 4c0n — CHANGES_REQUESTED** [review]

**Pull request merged**